test/issuance.spec.ts

```diff
@@ -0,0 +1,322 @@
+
+import { ECPair, networks } from '../ts_src';
+import { satoshiToConfidentialValue } from './../ts_src/confidential';
+import * as fixtures from './fixtures/issuance.json';
+
+const typeforce = require('typeforce');
+
+describe('Issuance', () => {
+  const fixture = fixtures.emptyContract;
+  const fixtureWithContract = fixtures.withContract;
+  const entropy31bytes = Buffer.from(
+    '2b73af1c9ae64a6903b3055361dd7b75082003a85374049982fc1e8f31b9a8',
+    'hex',
+  );
+  const prevout: issuance.OutPoint = {
+    txHash: Buffer.from(fixture.prevout.txHash, 'hex').reverse(),
+    vout: fixture.prevout.index,
+  };
+
+  describe('Issuance artifacts generation (entropy, asset value and token value)', () => {
+    describe('Entropy generation', () => {
+      it('should properly generate the entropy from a prevout point of the blockchain', () => {
+        const entropy = issuance.generateEntropy(prevout);
+        assert.strictEqual(entropy.toString('hex'), fixture.expectedEntropy);
+      });
+    });
+
+    describe('Asset calculation', () => {
+      it('should compute the asset value from an entropy previously generated', () => {
+        const asset = issuance.calculateAsset(
+          Buffer.from(fixture.expectedEntropy, 'hex'),
+        );
+        assert.strictEqual(
+          asset.reverse().toString('hex'),
+          fixture.expectedAsset,
+        );
+      });
+
+      it('should throw an error if the entropy has not a lenght of 32 bytes', () => {
+        assert.throws(() => issuance.calculateAsset(entropy31bytes));
+      });
+    });
+
+    describe('Token calculation', () => {
+      it('should compute the reissuance token value from an entropy previously generated', () => {
+        const token = issuance.calculateReissuanceToken(
+          Buffer.from(fixture.expectedEntropy, 'hex'),
+        );
+        assert.strictEqual(
+          token.reverse().toString('hex'),
+          fixture.expectedToken,
+        );
+      });
+
+      it('should throw an error if the entropy has not a lenght of 32 bytes', () => {
+        assert.throws(() => issuance.calculateReissuanceToken(entropy31bytes));
+      });
+    });
+  });
+
+  describe('Issuance object generation', () => {
+    function validate(i: Issuance): boolean {
+      try {
+        typeforce(types.Hash256bit, i.assetBlindingNonce);
+        typeforce(types.Hash256bit, i.assetEntropy);
+        typeforce(
+          types.oneOf(
+            types.ConfidentialValue,
+            types.ConfidentialCommitment,
+            types.BufferOne,
+          ),
+          i.assetAmount,
+        );
+        typeforce(
+          types.oneOf(
+            types.ConfidentialValue,
+            types.ConfidentialCommitment,
+            types.BufferOne,
+          ),
+          i.tokenAmount,
+        );
+        return true;
+      } catch (err) {
+        return false;
+      }
+    }
+
+    it('should create a correct Issuance object without an issuance contract', () => {
```

Comment:
this test is not actually checking that the created issuance does not have an issuance contract. We should enforce it by manually checking that iss.contract is undefined